src/common/components/repositories-home/index.js

```diff
@@ -19,16 +20,19 @@ const SNAP_POLL_PERIOD = (15 * 1000);
 
 class RepositoriesHome extends Component {
   fetchData(props) {
-    const { snaps } = props;
+    const { hasSnaps, snaps, entities } = props;
 
-    if (snaps.success) {
+    // check both success and fetching to avoid redirecting
+    // based on previous success
+    if (snaps.success && !snaps.isFetching) {
```

Comment:
Honestly I'm not sure right now. We would need to identify the different cases where we use these statuses and see how we can solve it.

I opened #625 to deal with that later.